.github/workflows/lint-test.yml

```diff
@@ -81,7 +81,7 @@ jobs:
 
       - name: Restore Poetry installation 📌
         id: poetry
-        uses: actions/cache@v3
+        uses: actions/cache@v4
```

Comment:
Unrelated, but the version of the `cache` action in these tasks was inconsistent with the version used in the other tasks, which was causing issues with Poetry.